projects/core/src/cart/facade/multi-cart.service.ts

```diff
@@ -70,15 +79,19 @@ export class MultiCartService {
     toMergeCartGuid?: string;
     extraData?: any;
   }): Observable<ProcessesLoaderState<Cart>> {
+    // to support creating multiple carts at the same time we need to use different entity for every process
+    // simple random uuid generator is used here for entity names
+    const freshCartId = this.generateFreshCartId();
```

Comment:
At the moment I think it is at the lowest level to solve all the problems. We need that id for the selector and moving it into actions would make that impossible.

However maybe in the future this might also by handy for a method such as `createNamedCart` that would also need temp entities